java/Cherepnin_Oleh/src/com/zoolatech/lecture2/tasks/_2/RobotController.java

```diff
@@ -0,0 +1,116 @@
+package com.zoolatech.lecture2.tasks._2;
+
+import java.util.Locale;
+import java.util.Scanner;
+
+public class RobotController {
+    private static Scanner scanner = new Scanner(System.in);
+    private Robot robot;
+
+    public void controlRobot() {
+        boolean flag = true;
+        showInstruction();
+        while (flag) {
+            System.out.println("Input operation:");
+            String action = scanner.next();
+            switch (action.toUpperCase(Locale.ROOT)) {
+                case "M" -> {
+                    System.out.println("Input the number of steps:");
+                    robot.moveForward(scanner.nextInt());
+                }
+                case "L" -> robot.turnLeft();
+                case "R" -> robot.turnRight();
+                case "V" -> robot.showPosition();
+                case "E" -> flag = false;
+                case "I" -> showInstruction();
+                default -> System.out.println("Incorrect operation \nTo view instruction press: 'I'");
+            }
+        }
+    }
+
+    private void showInstruction() {
+        System.out.println("To move forward press: 'M'\n" +
+                "To turn left press: 'L'\n" +
+                "To turn right press: 'R'\n" +
+                "To view current position press: 'V'\n" +
+                "To stop program press: 'E'\n" +
+                "To view instruction press: 'I'");
+    }
+
+    public void createRobot() {
+        String widthS = "width",
+                heightS = "height";
+        Direction direction;
+
+        System.out.println("Welcome to robot controller!!! \nLet`s set room for robot!");
+        int width = initRoomParameter(widthS);
+        int height = initRoomParameter(heightS);
+        String answer = askQuestion(1);
+
+        if (answer.equalsIgnoreCase("Y")) {
+            int x = setStartPoint(width, widthS);
+            int y = setStartPoint(height, heightS);
+            answer = askQuestion(2);
+
+            if (answer.equalsIgnoreCase("Y")) {
+                direction = setDirection();
+                this.robot = new Robot(width, height, x, y, direction);
```

Comment:
you could set a default value for the variable direction during the initialization and override it a user wants to change it - that would allow to use a same constructor for all cases